src/aro/Parser.zig

```diff
@@ -6946,7 +6976,30 @@ fn assignExpr(p: *Parser) Error!?Result {
 
     var is_const: bool = undefined;
     if (!p.tree.isLvalExtra(lhs.node, &is_const) or is_const) {
-        try p.errTok(.not_assignable, tok);
+        if (p.getNode(lhs.node, .decl_ref_expr)) |decl_ref| {
+            const var_name = p.tokSlice(decl_ref.name_tok);
+            const decl_tok = p.getNode(decl_ref.decl, .variable).?.name_tok;
+            try p.errStr(.assign_to_const_var, tok, try p.assignToConstVarStr(var_name, decl_ref.qt));
+            try p.errExtra(.declared_const_here, decl_tok, .{ .str = var_name });
+        } else if (p.getNode(lhs.node, .array_access_expr)) |array_access| {
+            if (p.getNode(array_access.base, .cast)) |arr_cast| {
+                if (p.getNode(arr_cast.operand, .decl_ref_expr)) |arr_cast_delc_ref| {
+                    const arr_ref_var_name = p.tokSlice(arr_cast_delc_ref.name_tok);
+                    const arr_ref_decl = p.getNode(arr_cast_delc_ref.decl, .variable).?.name_tok;
+                    try p.errStr(.assign_to_const_var, tok, try p.assignToConstVarStr(arr_ref_var_name, arr_cast_delc_ref.qt));
+                    try p.errExtra(.declared_const_here, arr_ref_decl, .{ .str = arr_ref_var_name });
+                }
+            }
+        } else if (p.getNode(lhs.node, .member_access_ptr_expr)) |member_access| {
+            if (p.getNode(member_access.base, .decl_ref_expr)) |memb_acc_decl_ref| {
+                const struct_name = p.tokSlice(memb_acc_decl_ref.name_tok);
+                const struct_decl = p.getNode(memb_acc_decl_ref.decl, .variable).?.name_tok;
+                try p.errStr(.assign_to_const_var, tok, try p.assignToConstVarStr(struct_name, memb_acc_decl_ref.qt));
+                try p.errExtra(.declared_const_here, struct_decl, .{ .str = struct_name });
+            }
+        } else {
+            try p.errTok(.not_assignable, tok);
```

Comment:
This case doesn't result in any error:
```c
void foo(const int *a) {
    (a + 1)[2] = 1;
}
```

Maybe break out of a block in the other cases instead of having a final else case.